front/app/utils/participationMethodUtils.tsx

```diff
@@ -1,39 +1,78 @@
+import React from 'react';
 import { ParticipationMethod } from 'services/participationContexts';
 import { IPhaseData } from 'services/phases';
 import { IProjectData } from 'services/projects';
+import clHistory from 'utils/cl-router/history';
+import { FormattedMessage } from './cl-intl';
+import messages from './participationMethodUtilsMessages';
 
 type ParticipationMethodConfig = {
   /** We currently have 2 UIs for admins to edit the form definition. This
    * defines which UI, if any, the method uses */
   formEditor: 'simpleFormEditor' | 'surveyEditor' | null;
+  onFormSubmission?: any;
+  getModalContent: any;
 };
 
 const ideationConfig: ParticipationMethodConfig = {
   formEditor: 'simpleFormEditor',
+  onFormSubmission: (_project, _ideaId, _idea) => {
+    clHistory.push({
+      pathname: `/ideas/${_idea.data.attributes.slug}`,
+      search: `?new_idea_id=${_ideaId}`,
+    });
+  },
+  getModalContent: () => {
+    return null;
```

Comment:
Where is the modal content shown after adding an idea in an ideation or budgeting context? I would expect it here, but probably it is implemented elsewhere.

Reply:
The modal content is hard-coded in the idea page at the moment. I haven't made any changes there so I haven't abstracted it to the participationMethodConfig. Should I do that in this PR as well?

Reply:
Wouldn't that be best? Otherwise we have two solutions. The hardcoded one and the one in the participationMethodConfig. How much work is it? And if you do not do it as part of this PR, when will we do it?